### PR TITLE
CXXCBC-564: allow to specify all operations in pillowfight workload

### PR DESCRIPTION
**Pull request by avsej**

### CODE REVIEW

**Review by DemetrisChr — APPROVED** [review]
LGTM (just left one minor comment)

File: tools/pillowfight.cxx [hunk 1]
```diff
@@ -434,52 +505,46 @@ class pillowfight_app : public CLI::App
...
-        bool should_check_known_keys{ false };
-        switch (opcode) {
-          case operation::get:
-            should_check_known_keys = hit_chance_for_get_ > dist(gen);
+
+      auto known_keys_distribution =
+        std::uniform_int_distribution<std::size_t>(0, known_keys.size() - 1);
+      for (std::size_t i = 0; i < operation_batch_size_; ++i) {
+        auto operation = operation_generator.next_operation();
+        std::string document_id = (operation != operation::cmd_insert || known_keys.empty())
```
Comment (DemetrisChr) [inline tools/pillowfight.cxx:522]: I think the condition should be `operation != operation::cmd_insert && !known_keys.empty()` here

**Review by DemetrisChr — CHANGES_REQUESTED** [review]
Re-reviewing - just noticed a possible issue

File: tools/pillowfight.cxx (hunk 1 above)
Comment (avsej) [reply tools/pillowfight.cxx]: Good catch, thank you

File: tools/pillowfight.cxx [hunk 2]
```diff
@@ -44,24 +46,118 @@ namespace cbc
...
-const std::size_t default_operations_limit{ 0 };
-const std::size_t default_key_value_batch_size{ 100 };
-const std::size_t default_query_batch_size{ 0 };
-const std::chrono::milliseconds default_batch_wait{ 0 };
+constexpr operation_weights default_operation_ratio{};
+constexpr std::size_t default_document_body_size{ 0 };
+constexpr std::size_t default_operations_limit{ 0 };
+constexpr std::size_t default_operation_batch_size{ 100 };
+constexpr std::chrono::milliseconds default_batch_wait{ 0 };
 const std::size_t default_number_of_keys_to_populate{ 1'000 };
```
[the remark this answers is not in the document]
Comment (avsej) [reply tools/pillowfight.cxx]: fixed

**Review by DemetrisChr — APPROVED** [review]

**Pull request merged**